### PR TITLE
[REVIEW] Updating sparse prims based on recent changes

### PR DESCRIPTION
**Pull request by cjnolet**
Pulling soe of @lowener's recent changes over to RAFT

### CODE REVIEW

File: cpp/include/raft/sparse/distance/l2_distance.cuh [hunk 1]
```diff
@@ -149,12 +144,40 @@ class l2_expanded_distances_t : public distances_t<value_t> {
...
+  : public l2_expanded_distances_t<value_idx, value_t> {
+ public:
+  explicit l2_sqrt_expanded_distances_t(
+    const distances_config_t<value_idx, value_t> &config)
+    : l2_expanded_distances_t<value_idx, value_t>(config) {}
+
+  void compute(value_t *out_dists) override {
+    l2_expanded_distances_t<value_idx, value_t>::compute(out_dists);
+    // Sqrt Post-processing
+    value_t p = 0.5;  // standard l2
```
Comment (divyegala) [inline cpp/include/raft/sparse/distance/l2_distance.cuh:168]: `constexpr` on this

File: cpp/include/raft/sparse/distance/lp_distance.cuh [hunk 2]
```diff
@@ -124,10 +124,32 @@ class l2_unexpanded_distances_t : public distances_t<value_t> {
...
+  : public l2_unexpanded_distances_t<value_idx, value_t> {
+ public:
+  l2_sqrt_unexpanded_distances_t(
+    const distances_config_t<value_idx, value_t> &config)
+    : l2_unexpanded_distances_t<value_idx, value_t>(config) {}
+
+  void compute(value_t *out_dists) {
+    l2_unexpanded_distances_t<value_idx, value_t>::compute(out_dists);
+    // Sqrt Post-processing
+    value_t p = 0.5;  // standard l2
```
Comment (divyegala) [inline cpp/include/raft/sparse/distance/lp_distance.cuh:142]: Same as above

**Review by divyegala — APPROVED** [review]
Just a couple of small comments, otherwise LGTM! Pre-approving

**Comment by cjnolet** [comment]
@gpucibot merge

**Review by dantegd — APPROVED** [review]

**Pull request merged**